### PR TITLE
some failing tests

### PR DESCRIPTION
**Pull request by atmnk**
This are some failing tests. Might need to fix the controller for them to pass.

### CODE REVIEW

**Comment by championswimmer** [comment]
omg tdd from the get go. 

well, I'll add tests only after there is CI to run them (github actions)

**Comment by atmnk** [comment]
@championswimmer Rebased the branch, now this repo has github action as well to run the test for this PR to merge. I would have fixed the code as well but still learning go and need to understand how controller level validations are done.

**Comment by championswimmer** [comment]
Thanks, but I am not that much of a TDD fanatic to include breaking tests as a separate commit and then the feature to go with in separate one 

Can you include the logic too (which will pass this test). 

P.S. The tests are correct btw.

File: tests/routes/api/users_test.go [hunk 1]
```diff
@@ -58,3 +58,18 @@ func TestUsersRoute_GetUserInfo(t *testing.T) {
...
+	req.Header.Set("Content-Type", "application/json; charset=UTF-8")
+	resp := lo.Must(app.Test(req))
+	assert.Equal(t, 400, resp.StatusCode)
+}
+func TestUsersRoute_ShouldNotRegisterUserWhenNoEmail(t *testing.T) {
+	reqBody := []byte(`{"password": "12345"}`)
+	req := httptest.NewRequest("POST", "/api/v1/users", bytes.NewBuffer(reqBody))
+	req.Header.Set("Content-Type", "application/json; charset=UTF-8")
+	resp := lo.Must(app.Test(req))
+	assert.Equal(t, 400, resp.StatusCode)
```
Comment (championswimmer) [inline tests/routes/api/users_test.go:73]: also, 422 better than 400 here

**Review by championswimmer — APPROVED** [review]

**Comment by championswimmer** [comment]
placeholder

**Pull request merged**